src/middleware/open-graph/utils.js

```diff
@@ -129,3 +131,100 @@ module.exports.getStatusCodes = (array) => {
 
 	return sortedCodes;
 };
+
+let networkData = Object.create(null);
+let lastNetworkDataRefresh = -1;
+
+let lastFetchSucceeded = false;
+const NETWORK_DATA_TTL = 1000 * 60 * 60;
+const SHORT_NETWORK_DATA_TTL = 1000 * 60;
+
+const fetchNetworkData = async () => {
```

Comment:
We already do this in the sitemap middleware and export that for other places too. Let's either use that or refactor it out to its own module in `src/lib` and use that everywhere.